vscode-extension/src/extension.ts

```diff
@@ -233,15 +227,27 @@ async function createNewAIConfig(
   const fileContentBuffer = await vscode.workspace.fs.readFile(fileContentPath);
   const initialContent = fileContentBuffer.toString();
 
-  const doc = await vscode.workspace.openTextDocument({
-    content: initialContent,
-    language: mode,
-  });
+  let doc: vscode.TextDocument = null;
+  let i = 0;
+  while (!doc) {
+    const fileName = `untitled${i === 0 ? "" : "-" + i}.aiconfig.${mode}`;
+    const filePath = workspaceUri
+      ? path.join(workspaceUri.path, fileName)
+      : fileName;
+    const untitledUri = vscode.Uri.file(filePath).with({ scheme: "untitled" });
+    try {
+      doc = await vscode.workspace.openTextDocument(untitledUri);
+      break;
+    } catch (err) {
+      i++;
+    }
+  }
+
+  const editor = await vscode.window.showTextDocument(doc, { preview: false });
 
-  //const doc = await vscode.workspace.openTextDocument(untitledUri);
-  await vscode.window.showTextDocument(doc, {
-    preview: false,
-    viewColumn: vscode.ViewColumn.One,
+  // TODO: saqadri - if the edits aren't applied, consider retrying or showing an error message
+  const editsApplied = await editor.edit((editBuilder) => {
```

Comment:
What does this `editsApplied` variable do? I don't see it used anywhere else? Is it a global var?